azure/scope/machinepool.go

```diff
@@ -486,15 +486,26 @@ func (m *MachinePoolScope) SetAnnotation(key, value string) {
 	m.AzureMachinePool.Annotations[key] = value
 }
 
-// PatchObject persists the machine spec and status.
+// PatchObject persists the AzureMachinePool spec and status.
 func (m *MachinePoolScope) PatchObject(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "scope.MachinePoolScope.PatchObject")
 	defer done()
 
-	return m.patchHelper.Patch(ctx, m.AzureMachinePool)
-}
-
-// Close the MachineScope by updating the machine spec, machine status.
+	conditions.SetSummary(m.AzureMachinePool)
+	return m.patchHelper.Patch(
+		ctx,
+		m.AzureMachinePool,
+		patch.WithOwnedConditions{Conditions: []clusterv1.ConditionType{
+			clusterv1.ReadyCondition,
+			infrav1.BootstrapSucceededCondition,
+			infrav1.ScaleSetDesiredReplicasCondition,
+			infrav1.ScaleSetModelUpdatedCondition,
+			infrav1.ScaleSetRunningCondition,
+			// TODO: Add a role assignment condition in a later PR
```

Comment:
Can you please file an issue for this instead of leaving the TODO in code? code TODOs tend to get forgotten...